MerlinAU.asp

```diff
@@ -517,216 +652,152 @@ function SetScheduleDAYofWEEK (cronDAYofWEEK)
+/** Modified by ExtremeFiretop [2025-Aug-24] **/
+/**------------------------------------------**/
+function FWConvertWebUISettingsToCronSchedule(oldRawCronSchedule){
+  const delimChar = '|';
+  const defaultSchedule = '0|0|*|*|*';
+
+  if (oldRawCronSchedule === 'TBD' || oldRawCronSchedule.split(' ').length < 5){
+    return defaultSchedule;
+  }
+
+  let fwRawSchedMINS = '0';
+  let fwRawSchedHOUR = '0';
+  let fwRawSchedDAYM = '*';
+  let fwRawSchedMNTH = '*';
+  let fwRawSchedDAYW = '*';
+
+  let T = document.getElementById('fwScheduleTIME');
+
+  // time from the single control
+  let parsed = ParseTimeHHMM(T ? T.value : '');
+  if (parsed.ok){
+    fwRawSchedHOUR = String(parsed.h);
+    fwRawSchedMINS = String(parsed.m);
+  }
+
+  // days logic (unchanged)
+  let fwScheduleDAYS1 = document.getElementById('fwSchedBoxDAYS1');
+  let fwSchedBoxDAYSX = document.getElementById('fwSchedBoxDAYSX');
+  let fwScheduleXDAYS = document.getElementById('fwScheduleXDAYS');
+  let fwScheduleMON = document.getElementById('fwSched_MON');
+  let fwScheduleTUE = document.getElementById('fwSched_TUE');
+  let fwScheduleWED = document.getElementById('fwSched_WED');
+  let fwScheduleTHU = document.getElementById('fwSched_THU');
+  let fwScheduleFRI = document.getElementById('fwSched_FRI');
+  let fwScheduleSAT = document.getElementById('fwSched_SAT');
+  let fwScheduleSUN = document.getElementById('fwSched_SUN');
+
+  if (fwScheduleDAYS1 && fwScheduleDAYS1.checked && fwScheduleDAYS1.disabled === false){
+    fwRawSchedDAYM = '*';
+    fwRawSchedDAYW = '*';
+  }
+  else if (fwSchedBoxDAYSX && fwSchedBoxDAYSX.checked && fwSchedBoxDAYSX.disabled === false){
+    if ((fwScheduleXDAYS.value == 2 || fwScheduleXDAYS.value == 3)){
+      fwRawSchedDAYM = '*';
+      fwRawSchedDAYW = '*/' + fwScheduleXDAYS.value;
+    } else {
+      fwRawSchedDAYW = '*';
+      fwRawSchedDAYM = '*/' + fwScheduleXDAYS.value;
+    }
+  }
+  else {
+    let daysOfWeekArray = [], daysOfWeekIndex = [];
+    if (fwScheduleSUN && fwScheduleSUN.checked && fwScheduleSUN.disabled === false){ daysOfWeekIndex.push(0); daysOfWeekArray.push('Sun'); }
+    if (fwScheduleMON && fwScheduleMON.checked && fwScheduleMON.disabled === false){ daysOfWeekIndex.push(1); daysOfWeekArray.push('Mon'); }
+    if (fwScheduleTUE && fwScheduleTUE.checked && fwScheduleTUE.disabled === false){ daysOfWeekIndex.push(2); daysOfWeekArray.push('Tue'); }
+    if (fwScheduleWED && fwScheduleWED.checked && fwScheduleWED.disabled === false){ daysOfWeekIndex.push(3); daysOfWeekArray.push('Wed'); }
+    if (fwScheduleTHU && fwScheduleTHU.checked && fwScheduleTHU.disabled === false){ daysOfWeekIndex.push(4); daysOfWeekArray.push('Thu'); }
+    if (fwScheduleFRI && fwScheduleFRI.checked && fwScheduleFRI.disabled === false){ daysOfWeekIndex.push(5); daysOfWeekArray.push('Fri'); }
+    if (fwScheduleSAT && fwScheduleSAT.checked && fwScheduleSAT.disabled === false){ daysOfWeekIndex.push(6); daysOfWeekArray.push('Sat'); }
```

Comment:
The above code lines are too long - over 100 chars each!!
This is also considered a bad coding practice.